akka-stream/src/main/scala/akka/stream/impl/JsonBracketCounting.scala

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (C) 2014-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import akka.stream.scaladsl.Framing.FramingException
+import akka.util.ByteString
+
+import scala.annotation.switch
+
+/**
+ * INTERNAL API: Use [[akka.stream.scaladsl.JsonFraming]] instead.
+ */
+private[akka] object JsonBracketCounting {
+
+  final val SquareBraceStart = '['.toByte
+  final val SquareBraceEnd = ']'.toByte
+  final val CurlyBraceStart = '{'.toByte
+  final val CurlyBraceEnd = '}'.toByte
+  final val DoubleQuote = '\''.toByte
+  final val Backslash = '\\'.toByte
+  final val Comma = ','.toByte
+
+  final val LineBreak = '\n'.toByte
+  final val LineBreak2 = '\r'.toByte
+  final val Tab = '\t'.toByte
+  final val Space = ' '.toByte
+
+  final val Whitespace = Set(LineBreak, LineBreak2, Tab, Space)
+
+  def isWhitespace(input: Byte): Boolean =
+    Whitespace.contains(input)
+
+}
+
+/**
+ * INTERNAL API: Use [[akka.stream.scaladsl.JsonFraming]] instead.
+ *
+ * **Mutable** framing implementation that given any number of [[ByteString]] chunks, can emit JSON objects contained within them.
+ * Typically JSON objects are separated by new-lines or comas, however a top-level JSON Array can also be understood and chunked up
+ * into valid JSON objects by this framing implementation.
+ *
+ * Leading whitespace between elements will be trimmed.
+ */
+private[akka] class JsonBracketCounting(maximumObjectLength: Int = Int.MaxValue) {
```

Comment:
Yeah, true (btw would love to merge this today it can be included in 2.4.9-RC1, please keep comments coming if you have the time!).

Netty recently introduced the same btw, and it's simply JsonDecorder https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/json/JsonObjectDecoder.java